renderdoc/driver/vulkan/vk_pixelhistory.cpp

```diff
@@ -1522,10 +1578,26 @@ struct VulkanPixelHistoryPerFragmentCallback : VulkanPixelHistoryCallback
 
     VkRenderPass newRp = CreateRenderPass(state.renderPass, state.GetFramebuffer(), state.subpass,
                                           m_Image, VK_FORMAT_R32G32B32A32_SFLOAT);
+
     VkFramebuffer newFb =
         CreateFramebuffer(state.renderPass, newRp, state.subpass, state.GetFramebuffer(),
                           m_StencilImageView, GetResID(m_Image), m_PerFragmentImageView);
-    Pipelines pipes = CreatePipelines(curPipeline, newRp, eid, false, 0);
+
+    const VulkanCreationInfo::Framebuffer &fbInfo =
+        m_pDriver->GetDebugManager()->GetFramebufferInfo(state.GetFramebuffer());
+    uint32_t framebufferIndex = 0;
+    for(uint32_t i = 0; i < fbInfo.attachments.size(); i++)
```

Comment:
Still one case here using the create-time images instead of the pipeline state's framebuffer attachment list (which will be either create-time or run-time images as appropriate).